packit_service/worker/handlers/distgit.py

```diff
@@ -791,6 +793,27 @@ def _get_or_create_koji_group_model(self) -> KojiBuildGroupModel:
     def get_branches(self) -> List[str]:
         """Get a list of branch (names) to be built in koji"""
 
+    def is_already_triggered(self, branch: str) -> bool:
+        """
+        Check if the build was already triggered
+        (building or completed state).
+        """
+        nvr = self.packit_api.dg.get_nvr(branch)
+        existing_build = KojiHelper().get_build_info(nvr)
```

Comment:
I'm not sure how expensive is instantiation of `koji.ClientSession` but it's probably worth keeping a single session open rather than creating a new one for every branch per build. We could use the same approach as in the past:

https://github.com/packit/packit-service/blob/6f028d40eb02100e5f6ae1d5c12123ddb5688c3e/packit_service/worker/handlers/distgit.py#L719-L744